onvif/examples/camera.rs

```diff
@@ -257,13 +262,19 @@ async fn get_stream_uris(clients: &Clients) {
             );
         }
     }
+    Ok(())
 }
 
-async fn get_snapshot_uris(clients: &Clients) {
-    let media_client = clients.media.as_ref().unwrap();
-    let profiles = schema::media::get_profiles(media_client, &Default::default())
-        .await
-        .unwrap();
+async fn get_snapshot_uris(clients: &Clients) -> Result<(), transport::Error> {
+    let media_client = match clients.media.as_ref() {
+        Some(client) => client,
+        None => {
+            return Err(transport::Error::Other(
+                "Client media is not available".into(),
+            ))
+        }
+    };
```

Comment:
This `match` can be replaced with `.ok_or(..)?` here and below.

Reply:
Probably `ok_or_else` since the `.into` allocates and clippy will complain otherwise.

Reply:
Good call!

Reply:
In the end I'm using `.ok_or` since it makes a bit more clear the code.

Reply:
Like I was saying, now clippy complains and I would tend to agree that its suggestion makes sense. Can you please apply them?